### DISCUSSION TITLE
client plugin reactivity

### DISCUSSION THREAD

**Discussion by djixadin** [nuxt/nuxt#22637] [Questions]
i made this simple plugin called theme.client.ts
```
export default defineNuxtPlugin(() => {
  const isDarkMode = ref(localStorage.theme === 'dark')

  if (isDarkMode.value || (!('theme' in localStorage) && window.matchMedia('(prefers-color-scheme: dark)').matches)) {
    isDarkMode.value = true
    document.documentElement.classList.add('dark')
  }

  const toggleTheme = () => {
    document.documentElement.classList.toggle('dark')
    isDarkMode.value = !isDarkMode.value

    if (isDarkMode.value) {
      localStorage.setItem('theme', 'dark')
    } else {
      localStorage.removeItem('theme')
    }
  }

  return {
    provide: {
      isDarkMode, toggleTheme
    }
  }
})
```

whenever i use this isDarkMode in a component on initial load its undefined but on every subsequent change reactivity is working but initially not. this can be fixed with a simple watch but is there a better way?

**Comment by TheAlexLichter** [comment]
because the `localStorage` is not available on the server side (where I assume the plugin is executed first).
You might want to check https://github.com/nuxt-modules/color-mode

**Reply by djixadin** [reply]
the plugin is a client plugin so the value in the vue component is undefined which makes sense but it should update on mounted but it does not. and im using tailwind already so dont want to include another package

**Reply by TheAlexLichter** [reply]
I see. Do you have an example stackblitz by any chance?

**Reply by djixadin** [reply]
https://stackblitz.com/edit/nuxt-starter-gyd4he?file=plugins%2Ftheme.client.ts

here you go. managed it

**Reply by TheAlexLichter** [reply]
Nice. Don't forget to actually call the helper only on client-side then or stub it on the server side 😋

**Reply by djixadin** [reply]
so this is intended behavior? i need ```<ClientOnly>``` as a wrapper?

**Reply by xjccc** [reply]
```
<template>
  <div>
    <button @click="$toggleTheme">
      {{ $isDarkMode ? 'light' : 'dark' }}
    </button>
  </div>
</template>
<script setup>
const {$toggleTheme, $isDarkMode} = useNuxtApp()
</script>
```
